### PR TITLE
[luajit] update to the latest commit

### LINKED ISSUE #25851: [luajit] add 2.1.0-beta3
Library name: luajit

New version number: 2.1.0-beta3

Many people are using the v2.1 branch of https://github.com/LuaJIT/LuaJIT and it would be useful to have that as an option, even if it's not the default

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fix #25851 

Update luajit to the latest commit

Note: no feature need to test

### CODE REVIEW

**Comment by JonLiu1993** [comment]
I will wait for the 2.1.0 stable version to be released and continue to upgrade to the latest version and then merge pr

**Comment by BurningEnlightenment** [comment]
Whouldn't it be possible to add the beta version and leave the baseline untouched? That way people could explicitly opt-in via versioning.

**Comment by JonLiu1993** [comment]
> Whouldn't it be possible to add the beta version and leave the baseline untouched? That way people could explicitly opt-in via versioning.

If you want to use this beta, just use my pr, you can also ask upstream if they will release the latest version soon, because I found that they are still submitting pr upstream to solve the problem recently

**Comment by FrankXie05** [comment]
Related upstream issues:  https://github.com/LuaJIT/LuaJIT/issues/563

**Review by FrankXie05 — APPROVED** [review]

File: ports/luajit/portfile.cmake [hunk 1]
```diff
@@ -1,8 +1,8 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO LuaJIT/LuaJIT
-    REF v2.0.5
-    SHA512 65d982d7fe532a61335613f414f3b8fa5333747bdf7aefc2c2d52022d227594ade827639049b97e3c4ffae9f38f32cb15f1a17b1780fb0a943e1a3af05e2b576
+    REF 633f265f67f322cbe2c5fd11d3e46d968ac220f7  #2022-08-11
```
Comment (FrankXie05) [inline ports/luajit/portfile.cmake:4]: According to the upstream reply(https://github.com/LuaJIT/LuaJIT/issues/563#issuecomment-1208958383), we decided to update `luajit` to the latest [release-time](https://github.com/LuaJIT/LuaJIT/commit/633f265f67f322cbe2c5fd11d3e46d968ac220f7).

**Comment by BillyONeal** [comment]
Thanks!

**Comment by wsw0108** [comment]
I think the 'luajit.lib' should also be installed.

**Comment by zao** [comment]
This PR as merged doesn't actually build or install LuaJIT. The `minilua` that it installs is a build-time tool used for code generation, the actual LuaJIT libraries are not successfully built nor installed.

The build bails out very early being provided an incorrect path to `vm_*.dasc`.
If you fix that then the build of LuaJIT itself is broken in that the generated source files do not seem to generate like they do with unmodified upstream source and fails the build again due to this.

These problems are likely due to the separation of the build directory from source directory, causing the generators to not find all the source.

**Comment by zao** [comment]
I've managed to build this properly now by undoing most of the changes in this PR. Should one open a new PR to fix the state of this port in vcpkg?

`001-fix-build-path.patch`:
`ALL_LIB` needs to include `%SOURCEDIR%\lib_buffer.c`, the current patch accidentally removed that source file from the variable.

The command to generate `host\buildvm_arch.h` needs to point out the correct path for both the generated header and `%DASC%`. If it's unacceptable to generate data in the source tree, we'll instead need to create the target directory and augment include search paths to find it.

```patch
-minilua %DASM% -LN %DASMFLAGS% -o host\buildvm_arch.h %DASC%
+minilua %DASM% -LN %DASMFLAGS% -o %SOURCEDIR%\host\buildvm_arch.h %SOURCEDIR%\%DASC%
```

`portfile.cmake`:
This file needs to revert all the changes around `INSTALL` files, the proper files to install are as before `luajit.exe`, `lua51.lib` and `lua51.dll`; in a correct build `minilua` files are deleted as the build concludes.

**Comment by BillyONeal** [comment]
> I've managed to build this properly now by undoing most of the changes in this PR. Should one open a new PR to fix the state of this port in vcpkg?

Done, I think :) https://github.com/microsoft/vcpkg/pull/26360

Thanks for the report.

**Pull request merged**